src/apis.cpp

```diff
@@ -86,33 +86,57 @@ static void gds_init_wait_request(gds_wait_request_t *request, uint32_t offset)
 }
 
 //-----------------------------------------------------------------------------
-
-int gds_post_send(struct gds_qp *qp, struct ibv_send_wr *wr, struct ibv_send_wr **bad_wr)
+int gds_rollback_qp(struct gds_qp *qp, gds_send_request_t * send_info, int flags)
```

Comment:
flags suggests a bit mask, here you should use enum ibv_exp_rollback_flags and make it static